### PR TITLE
Add missing `@selector` for Objective-C methods

### PR DESCRIPTION
**Pull request by jacob-carlborg**

### CODE REVIEW

**Comment by jacob-carlborg** [comment]
@kinke

File: test/dshell/extra-files/cpp_header_gen/library.d [hunk 1]
```diff
@@ -113,7 +116,8 @@ extern(C++) class VTable
 {
     extern(D) int hidden_1() { return 1; }
     int callable_2() { return 2; }
-    extern(Objective-C) int hidden_3() { return 3; }
+    version (D_ObjectiveC)
+        extern(Objective-C) int hidden_3() @selector("hidden_3") { return 3; }
```
Comment (MoonlightSentinel) [inline test/dshell/extra-files/cpp_header_gen/library.d:120]: You could change the linkage to `extern(D)` instead. The test only verifies that the header generator emits functions with appropriate linkage (and doesn't screw with the vtable layout).

**Review by MoonlightSentinel — APPROVED** [review]

File: test/dshell/extra-files/cpp_header_gen/library.d (hunk 1 above)
Comment (jacob-carlborg) [reply test/dshell/extra-files/cpp_header_gen/library.d:120]: Well, it was `Objective-C` from the beginning and I'm lazy 😃.

**Comment by kinke** [comment]
Thanks! Please target stable to slightly simplify things for me. ;)

**Comment by jacob-carlborg** [comment]
Done.

**Review by kinke — APPROVED** [review]

**Pull request merged**